### DISCUSSION TITLE
programming ergonomic: Add Vim navigation to Code Actions list menu

### DISCUSSION THREAD

**Discussion by hajarrashidi** [zed-industries/zed#7983] [Feature Requests]
Is it possible to add Vim navigation to Code Actions list menu. To be able to go up and down with vim keys 
**J** key for going down and **K** for going up.

What it solves: You don't have to move your hand away from home row. 
Current issue: You have to leave your right hand from home row and instead use arrow keys.

<img width="779" alt="Screenshot 2024-02-18 at 17 16 58" src="https://github.com/zed-industries/zed/assets/4338882/a93af4ac-decc-4f1b-9903-723975c2855b">

**Comment by 0x2CA** [comment]
I now use c-p, c-n, c-f, c-b
Of course it is better to use c-h c-j c-k c-l

**Comment by mrnugget** [comment]
You can do this:

```json

  {
    "context": "Editor && vim_mode == normal && menu",
    "bindings": {
      "j": "editor::ContextMenuNext",
      "k": "editor::ContextMenuPrev"
    }
  },
```

**Reply by okbytes** [reply]
Non-deprecated settings for Zed post-March 2025:
```json
{
  "context": "Editor && vim_mode == normal && menu",
  "bindings": {
    "j": "editor::ContextMenuNext",
    "k": "editor::ContextMenuPrevious"
  }
}
```
